pelican/log.py

```diff
@@ -1,142 +1,446 @@
 import logging
 from collections import defaultdict
+from contextlib import contextmanager
 
 from rich.console import Console
 from rich.logging import RichHandler
 
-__all__ = [
-    'init'
-]
 
 console = Console()
 
+# handles messages logged from anywhere within the pelican namespace.
+PKG_LOGGER = logging.getLogger('pelican')
+
+
+def severity_from_name(string_val):
+    """Convert string value to appropriate severity (AKA log level)."""
+    if not string_val:
+        return logging.NOTSET
+    level_name = string_val.upper().rstrip('S')
+    try:
+        return logging.getLevelNamesMapping()[level_name]
+    except AttributeError:  # Python < 3.11
+        return logging._nameToLevel.copy()[level_name]
```

Comment:
The problem with ``getLevelName`` is that it's intended to take an ``int`` and return a ``str``, but I need the opposite conversion.

While its *technically* possible to muscle this behavior out of this ``logging.getLevelName``
1.  the name is misleading and counter to the intended effect
2.  the documentation basically says this behavior was accidental, removed, and then added back because it was breaking old code.  In short -- using it is relying on a hack.  

The preferred way of doing the conversion (``logging.getLevelNamesMapping``) is used instead -- but as this isn't available in Python < 3.11 I'm using the alternative as shown above, when necessary.

The other reason I don't use ``getLevelName``  is that it can potentially return a string, and I need to guarantee an ``int``.

I'd rather keep this as-is.

Reply:
Yes it is accidental (hence why I said "unintended") and maaybe considered a hack... But it is documented and is far more likely to remain as is in a minor patch than an undocumented internal dict. That's why I suggested :). I don't feel that strongly about it, thus I can live with it as is. Why do you `.copy()` it though?

Reply:
I'm just imitating what the standard lib does:

```python
# logging.py
def getLevelNamesMapping():
    return _nameToLevel.copy()
```
I'm guessing its there to future proof against an accidental pass by reference?

Reply:
Ah OK. Yeah, it's probably copying so that any modification on the returned `dict` doesn't affect the original version. It doesn't seem necessary here.